### ISSUE TITLE
Tree with nested nodes example doesn't use valid HTML structure

### ISSUE THREAD

**Issue by benhurqc** [angular/components#22517]
This is an a11y issue reported by many a11y tools. When using mat-tree component, li elements are introduced in mat-nested-tree-node elements when they should be in an ul or ol element. When trying to change mat-tree component elements by ul there's conflicts because of role="tree". Structure needs to be revised so listitem elements are direct children of list elements and elements with role="treeitem" are direct children of elements with role="tree".

**Comment by jelbourn** [comment]
Confirmed that the "Tree with nested nodes" example doesn't use the right list DOM structure

**Comment by walvekarnikhil** [comment]
Can I work on this?
How can we test this once fixed?

**Comment by benhurqc** [comment]
Personally I test using Axe for accessibility for the proper structure.

**Comment by jelbourn** [comment]
As far as I know this should only affect the example, not the tree component itself. Feel free to send a PR if you want to take a pass at it.
